tree/treeplayer/src/TSimpleAnalysis.cxx

```diff
@@ -0,0 +1,286 @@
+      }
+   }
+}
+
+
+////////////////////////////////////////////////////////////////////////////////
+/// Tell if the name of some histograms that will be created are the same
+///
+/// param[in] name name of the histograms
+
+void TSimpleAnalysis::CheckHNames(std::vector<std::string> name)
+{
+   Int_t err=0;
+   for (unsigned i=0; i<name.size()-1; i++)
+      for (unsigned j=i+1; j<name.size(); j++) {
+         if (name[i] == name[j]) {
+            ::Error("TSimpleAnalysis::check_name",
+                    "Multiple %s in the names of histograms (position %d and %d)",
+                    name[i].c_str(), i, j);
+            err++;
+         }
+      }
+   if (err != 0)
+      throw 1;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Return true if the input arguments create the output file correctly
+
+Bool_t TSimpleAnalysis::Analysis()
+{
+   TChain chain(fTreeName.c_str());
+   for (const std::string& inputfile: fInputFiles)
+      chain.Add(inputfile.c_str());
+   TFile ofile(fOutputFile.c_str(),"RECREATE");
+
+   CheckHNames(fHNames);
+
+   for (unsigned i=0; i<fExpressions.size(); i++) {
+      chain.Draw((fExpressions[i] + ">>" + fHNames[i]).c_str(),fCut[i].c_str(),"goff");
+      TH1F *histo = (TH1F*)gDirectory->Get(fHNames[i].c_str());
+      histo->Write();
+   }
+
+   return true;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Return false if not a tree name
+///
+/// param[in] line line read from the input file
+
+Bool_t TSimpleAnalysis::HandleTreeNameConfig(const std::string& line)
+{
+   if (line.find("=") != std::string::npos)
+      return false;
+   else return true;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// It handles the expression lines of the input file in order to pass the
+/// elements to the members of the object
+///
+/// param[in] line line read from the input file
+/// param[in] numbLine number of the input file line
+
+void TSimpleAnalysis::HandlefExpressionConfig(std::string& line, int& numbLine)
+{
+   std::size_t equal=line.find("=");
+   if (equal == std::string::npos) {
+      ::Error("TSimpleAnalysis::HandlefExpressionConfig",
+              "Missing '=' in fExpressions in %s:%d",fInputName.c_str(), numbLine);
+      throw 1;
+   }
+   std::size_t cutPos=line.find(kCutIntr, equal);
+   std::string substring=line.substr(0,equal);
+   if (substring.empty()) {
+      ::Error("TSimpleAnalysis::HandlefExpressionConfig",
+              "No hname found in %s:%d",fInputName.c_str(), numbLine);
+      throw 2;
+   }
+   fHNames.push_back(substring);
+   substring=line.substr(equal+1,cutPos-equal-1);
+   if (substring.empty()) {
+      ::Error("TSimpleAnalysis::HandlefExpressionConfig",
+              "No expression found in %s:%d",fInputName.c_str(), numbLine);
+      throw 3;
+   }
+   fExpressions.push_back(substring);
```

Comment:
For using `map<string /*name*/, string /*expr*/> fHists;` you don't use `substring` but `name` and `expr` as string variables, and then use std::map::insert (see http://www.cplusplus.com/reference/map/map/insert/) which allows you to check whether a histogram with that name already exists.